src/test/java/de/poiu/apron/io/PropertyFileReaderTest.java

```diff
@@ -425,6 +426,67 @@ public void testSingleBackslashOnLine() throws IOException {
     assertThat(readEntries.get(1)).isEqualTo(new PropertyEntry("   ", "\\ ", "   ", "", "\n"));
   }
 
+  /**
+   * This test verifies bug #11: https://github.com/hupfdule/apron/issues/11
+   * @throws IOException
+   */
+  @Test
+  public void test_CrLfEndings() throws IOException {
+    // - preparation
+    final File propertyFile= this.createTestFile(""
+        + "keyA1=valueA1\n"
+        + " keyA2  =  valueA2\n"
+        + "\tkeyA3\t=\tvalue A3\t\n"
+        + "keyA4 = very long\\\r\n"
+        + "value A4 over \\\r\n"
+        + "multiple lines\r\n"
+        + "        \n"
+        + "keyB1:valueB1\n"
+        + " keyB2 : valueB2\n"
+        + "\t keyB3\t:\t value B3 \n"
+        + "keyB4 : very long\\\n"
+        + "value B4 over \\\n"
+        + "multiple lines\\\n"
+        + "\n"
+        + "keyC1 valueC1\n"
+        + "  keyC2   valueC2\n"
+        + "\t keyC3\t\tvalue C3 \n"
+        + "keyC4   very long\\\n"
+        + "value C4 over \\\n"
+        + "\t \tmultiple lines");
+
+    // - execution
+    final List<String> readEntries= new ArrayList<>();
+    try (final PropertyFileReader reader= new PropertyFileReader(propertyFile);) {
+      Entry entry;
+      while ((entry= reader.readEntry()) != null) {
+        readEntries.add(entry.toCharSequence().toString());
+      }
+    }
+
+    // - validation
+    assertThat(readEntries.size()).isEqualTo(13);
+    assertThat(readEntries.get(0)).isEqualTo("keyA1=valueA1\n");
+    assertThat(readEntries.get(1)).isEqualTo(" keyA2  =  valueA2\n");
+    assertThat(readEntries.get(2)).isEqualTo("\tkeyA3\t=\tvalue A3\t\n");
+    assertThat(readEntries.get(3)).isEqualTo("keyA4 = very long\\\r\n" +
+        "value A4 over \\\r\n" +
+        "multiple lines\r\n");
+    assertThat(readEntries.get(4)).isEqualTo("        \n");
```

Comment:
@hupfdule I'm not entirely sure about this line. It contains only white-space. It depend on what do you expect from `readLogicalLine` output. It might be correct but also not as well.

Reply:
A whitespace line is a separate Entry of type BasicEntry. A `java.util.Properties` would just drop this empty line. As Apron retains all formatting (which includes empty lines) this would be a separate entry. However, we don't need to test this here as we only want to test the handling of the escaped `\r\n`. 

Also you would need to compare the read entries to the correct `de.poiu.apron.entry.PropertyEntry` instances. Again, see the other tests that already do this.

An alternative is to not include this test here, but instead write a test in `de.poiu.apront.PropertyFileTest`. which executes some sort of end-to-end test (actually ignoring the details of parsing and only checking the resulting key-value-pairs). As we do not test any BasicEntries this would be enough. The preparation steps are the same. Only the execution and verification steps would usually be:

```java
    // - execution
    final PropertyFile readPropertyFile= PropertyFile.from(propertyFile);

    // - validation
    assertThat(readPropertyFile.propertiesSize()).isEqualTo(javaUtilProperties.size());
    for (final Map.Entry<Object, Object> e : javaUtilProperties.entrySet()) {
      final String key = (String) e.getKey();
      final String value = (String) e.getValue();
      assertThat(readPropertyFile.containsKey(key)).as("key %s contains value %s", key, value).isTrue();
    }
```

It's up to you whether you want to leave this test here and update it accordingly or move it to such an end-to-end test in `PropertyFileTest`.